spec/index.adoc

```diff
@@ -2406,9 +2406,16 @@ Conditions::
     ) ∨ (
       E.content.canister_id ≠ ic_principal
       S.canisters[E.content.canister_id] ≠ EmptyCanister
-      Arg = { data = E.content.arg; caller = E.content.sender; method = E.content.method_name; time = S.time[E.content.canister_id] }
+      Arg = { data = E.content.arg; caller = E.content.sender }
+      Env = {
+        time = S.time[E.content.canister_id];
+        balance = S.balances[E.content.canister_id]
+        freezing_limit = freezing_limit(S, E.content.canister_id);
+        certificate = NoCertificate;
+        status = S.status[E.content.canister_id];
```

Comment:
I believe you mean `canister_status` here?

Reply:
Yes, probably. I copied this from elsewhere, so it was wrong there too. I'll refine.

Reply:
Also, the function `freezing_limit` is not defined anywhere in the master version of `index.adoc`.

Reply:
@nomeata No, it can't be the `CanStatus`, because `Env` specifies:
```status : Running | Stopping | Stopped```
Is this perhaps a reference to the `CanStatus` that is later defined?
```
CanStatus
  = Running
  | Stopping (List (CallOrigin, Nat))
  | Stopped
```
Or are they separate types that get lined up in the state transition?

Reply:
Ah, yes, that’s a bit annoying. In `S` we want to keep track of the pending call contexts when the state is `Stopping`, but in `Env` (and also in the Canister status) it is just an enum. Right now we are type-wise sloppy here.

Reply:
Let's introduce a type conversion function between `CanStatus` and `status : Running | Stopping | Stopped` (it's called `to_status` in the formal model).

Reply:
I've introduced the conversion function and named it `simple_status` (both in textual and formal spec).